2.5.0

CraZyRc opened the pull request.

CraZyRc committed:
Commands:
- added translator onReload() to the reload command

Translator:
- added loadFiles() (to create the language folder/files)
- added onReload() (to reload message changes)
- changed LookupTranslation() to read from file instead of resource

GnarLito commented on src/main/java/whackamole/whackamole/Translator.java:
```diff
@@ -258,8 +266,39 @@ private static String typesToString(Object[] types) {
+
+        // * Files creation
+        try {
+            for (String language : languages) {
```
what if we want to add a new item for translation? This will only run once to create the file and then never again

GnarLito commented on src/main/java/whackamole/whackamole/Translator.java:
```diff
@@ -172,9 +176,13 @@ private Translator(String key, Object... requiredTypes) {
         try {
-            ResourceBundle resource = ResourceBundle.getBundle("Lang", Config.AppConfig.Language);
-            this.value = resource.getString(this.key);
+            FileInputStream getProps = new FileInputStream(langFile);
```
Reading the whole file every time is a waste. This should only be done at `onLoad` or `onReload`

GnarLito replied:
I stand corrected, this method is already only running on `onLoad` and `onReload`

GnarLito reviewed (commented):
I do see some problems with the new way of working with language files

CraZyRc committed:
language.properties:
- added Translator property

Translator:
- changed the Translation lookup
- added concept for fileUpdater

GnarLito committed:
Start refactor ResourceManager

GnarLito committed:
Add section support

GnarLito committed:
Allow new section at the beginning

GnarLito committed:
Allow for multiple sections to be inserted at once

GnarLito committed:
Move TestBase to helpers

GnarLito committed:
Remove Config tests

GnarLito committed:
Fix logging not working from resource manager

GnarLito committed:
Fix resource loading

GnarLito committed:
Remove unneeded depenency

CraZyRc committed:
Commands:
- fixed reload command to fix language switch after wam reload with another language set in the config
- removed "whack" alias, since "whackamole" and "wam" are enough

Config:
- removed "HIDE_ATTRIBUTES" flag from restart ticket and wam hammer for 1.20.6 compatibility

Game:
- fixed particle type from "REDSTONE" to "DUST"

pom.template:
- updated to latest dependency builds

CraZyRc committed:
added fix for UTF-8 reading in 1.20.6+

CraZyRc committed:
added UTF-8 encoding for all supported mc versions (1.18.1 - 1.21)
